crates/pop-common/src/sourcing/binary.rs

```diff
@@ -112,10 +112,12 @@ impl Binary {
 					path.exists().then_some(Some(version.to_string()))
 				})
 				.nth(0)
-				.unwrap_or(
+				.unwrap_or_else(|| {
 					// Default to latest version
-					available.first().map(|version| version.as_ref().to_string()),
-				),
+					let versions = available.iter().map(|v| v.as_ref()).collect::<Vec<&str>>();
+					let latest = parse_latest_tag(versions);
```

Comment:
This function has also now gone from being clean and specific to doing one thing, to knowing about parsing SDK tags making testing harder. Rather than this function directly coupling to `parse_latest_tag`, it should perhaps instead take a `resolver` parameter which allows the caller to specify the `parse_latest_tag` function.

I expect this is too much to do under the time constraints, so happy for an issue to be created to correct this later.

Reply:
Added into our internal sprint planning